### PR TITLE
fix: pickle compatible implementation of downsamplers

### LINKED ISSUE #42: fix: support serialization with pickle
Currently none of the Rust downsamplers can be serialized with pickle as pickle throws a TypError when serializing a module

- [ ] add support for pickle
- [ ] add tests for this

Minimal code example :arrow_down: 

```py
In [1]: from tsdownsample import MinMaxDownsampler

In [2]: d = MinMaxDownsampler()

In [3]: import pickle

In [4]: pickle.dump(d, open('test.pkl', 'wb'))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[4], line 1
----> 1 pickle.dump(d, open('test.pkl', 'wb'))

TypeError: cannot pickle 'module' object

```

---
Related: PR #34 added support + tests for `deepcopy`

### PR DESCRIPTION
**Pull request by jonasvdd**
Closes #42 

TODO:
- [x] tests
- [x] code reveiw

### CODE REVIEW

File: tsdownsample/downsampling_interface.py [hunk 1]
```diff
@@ -143,24 +143,28 @@ def downsample(self, *args, n_out: int, **kwargs):  # x and y are optional
 class AbstractRustDownsampler(AbstractDownsampler, ABC):
     """RustDownsampler interface-class, subclassed by concrete downsamplers."""
 
-    def __init__(self, resampling_mod: ModuleType):
-        super().__init__(_rust_dtypes, _y_rust_dtypes)  # same for x and y
-        self.rust_mod = resampling_mod
+    @property
+    def rust_mod(self):
```
Comment (jonasvdd) [inline tsdownsample/downsampling_interface.py:147]: maybe add a line of documentation here?

File: tsdownsample/downsampling_interface.py [hunk 2]
```diff
@@ -143,24 +143,28 @@ def downsample(self, *args, n_out: int, **kwargs):  # x and y are optional
...
         if hasattr(self.rust_mod, "simd"):
             # use SIMD implementation if available
-            self.mod_single_core = self.rust_mod.simd
+            return self.rust_mod.simd
+        return self.rust_mod.scalar
 
-        # Store the multi-core sub module (if present)
-        self.mod_multi_core = None  # no multi-core implementation (default)
+    @property
+    def mod_multi_core(self):
```
Comment (jonasvdd) [inline tsdownsample/downsampling_interface.py:158]: maybe also add type hint that we return none, when no parallel implementation is available?

File: tsdownsample/downsampling_interface.py [hunk 3]
```diff
@@ -143,24 +143,28 @@ def downsample(self, *args, n_out: int, **kwargs):  # x and y are optional
 class AbstractRustDownsampler(AbstractDownsampler, ABC):
     """RustDownsampler interface-class, subclassed by concrete downsamplers."""
 
-    def __init__(self, resampling_mod: ModuleType):
-        super().__init__(_rust_dtypes, _y_rust_dtypes)  # same for x and y
-        self.rust_mod = resampling_mod
+    @property
```
Comment (jonasvdd) [inline tsdownsample/downsampling_interface.py:146]: do we put the properties above or below the `__init__`

Comment (jvdd) [reply tsdownsample/downsampling_interface.py:146]: I would do below the __init__

**Pull request merged**